src/fluree/db/json_ld/credential.cljc

```diff
@@ -62,51 +59,64 @@
   an exception will be thrown."
   [credential]
   (go-try
-    (when-let [jws (get-in credential ["proof" "jws"])]
-      (let [subject (get credential "credentialSubject")
-            {:keys [header signature]} (deserialize-jws jws)
-
-            signing-input #?(:clj (-> (jld-processor/canonize subject)
-                                      (crypto/sha2-256))
-                             :cljs (<p! (-> (jld-processor/canonize subject)
-                                            (.then (fn [res] (crypto/sha2-256 res))))))
-
-            proof-did     (get-in credential ["proof" "verificationMethod"])
-            pubkey        (did/decode-did-key proof-did)
-            id            (crypto/account-id-from-public pubkey)
-            auth-did      (did/auth-id->did id)]
-        (when (not= jws-header-json header)
-          (throw (ex-info "Unsupported jws header in credential."
-                          {:error :credential/unknown-signing-algorithm
-                           :supported-header jws-header-json
-                           :header header
-                           :credential credential})))
-
-        (when (not (crypto/verify-signature pubkey signing-input signature))
-          (throw (ex-info "Verification failed." {:error :credential/invalid-signature :credential credential})))
+   (when-let [jws (get-in credential ["proof" "jws"])]
+     (let [subject (get credential "credentialSubject")
+           {:keys [header signature]} (deserialize-jws jws)
+
+           signing-input #?(:clj (-> (jld-processor/canonize subject)
+                                     (crypto/sha2-256))
+                            :cljs (<p! (-> (jld-processor/canonize subject)
+                                           (.then (fn [res] (crypto/sha2-256 res))))))
+
+           proof-did     (get-in credential ["proof" "verificationMethod"])
+           pubkey        (did/decode-did-key proof-did)
+           id            (crypto/account-id-from-public pubkey)
+           auth-did      (did/auth-id->did id)]
+       (when (not= jws-header-json header)
+         (throw (ex-info "Unsupported jws header in credential."
+                         {:error :credential/unknown-signing-algorithm
+                          :supported-header jws-header-json
+                          :header header
+                          :credential credential})))
+
+       (when (not (crypto/verify-signature pubkey signing-input signature))
+         (throw (ex-info "Verification failed." {:error :credential/invalid-signature :credential credential})))
         ;; everything is good
-        {:subject subject :did auth-did}))))
+       {:subject subject :did auth-did}))))
+
+(defn verify-jws
+  [jws]
+  (let [{:keys [payload pubkey]} (crypto/verify-jws jws)
+        id                       (crypto/account-id-from-public pubkey)
+        auth-did                 (did/auth-id->did id)]
+    {:subject (json/parse payload false) :did auth-did}))
```

Comment:
Hardcoding `json/parse` of the payload precludes the use of edn strings in the payload. This can be fixed in the future by allowing users to specify the header for their JWS and the content-type for the payload. Since that adds a lot of complexity to the implementation that I'm not sure that we'll actually need I've not supported that here.

Reply:
We've done a lot of work to make fluree feel natural in Clojure, and Clojure developers will be one of the first populations we focus on early in the new year. We should at the very least have a ticket for enabling edn strings.

Reply:
I agree, I've created https://github.com/fluree/core/issues/66 to track it.